File: 0275-h-index-ii/0275-h-index-ii.java
class Solution {
    public int hIndex(int[] citations) {
        
        int left = 0, n = citations.length, right = n-1, mid;
        int h_index = 0;
        
        while (left <= right){
            mid = left + (right - left)/2;
            
            if(citations[mid] >= n - mid){
                h_index = n - mid;
                right = mid - 1;
            }
            else{
                left = mid + 1;
            }
        }
        
        return h_index;
        
        
    }
}

File: 0275-h-index-ii/0275-h-index-ii.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        
        left, right = 0, len(citations)-1
        h_index = 0
        
        while left <= right:
            
            mid = (left + right)//2
            
            pappers = len(citations) - mid
            
            if citations[mid] >= pappers:
                h_index = pappers
                right = mid - 1
            else:
                left = mid + 1
        
        return h_index